packages/workspace/src/browser/workspace-service.ts

```diff
@@ -318,11 +318,11 @@ export class WorkspaceService implements FrontendApplicationContribution {
     /**
      * Clears current workspace root.
      */
-    close(): void {
+    async close(): Promise<void> {
         this._workspace = undefined;
         this._roots.length = 0;
 
-        this.server.setMostRecentlyUsedWorkspace('');
```

Comment:
Maybe we could talk on Monday in person. Like Simon, I don't really understand the difference.

Reply:
I had to make sure to `await` the call `this.server.setMostRecentlyUsedWorkspace('')` in order for the backend to properly finish updating before Theia can reload. What I noticed in Firefox for example is that the update never resolves so when Theia reloads it reloads with the workspace that was opened before the `Close Workspace` was called.